packages/2018-housing-affordability/src/state/housing-policy/selectors.js

```diff
@@ -0,0 +1,96 @@
+import { createSelector } from 'reselect';
+import { groupBy } from 'lodash';
+import { rootState } from '../selectors';
+
+export const getHousingPolicy = createSelector(
+  rootState,
+  ({ housingPolicy }) => housingPolicy
+);
+
+const addLinks = governments => governments.map(program => ({
+  ...program,
+  links: [
+    { link: program.link1, link_name: program.link1_name },
+    { link: program.link2, link_name: program.link2_name },
+    { link: program.link3, link_name: program.link3_name },
+  ],
+}));
+
+const groupByEntity = governments => groupBy(
+  governments,
+  item => item.government_entity
+);
+
+const getProperty = key =>
+  createSelector(getHousingPolicy, state => state[key]);
+
+export const isLoading = getProperty('pending');
+export const isError = getProperty('error');
+export const getAllPolicies = getProperty('allPolicies');
+export const getAllPrograms = getProperty('allPrograms');
+export const getSelectedPolicy = getProperty('selectedPolicy');
+
+export const getTableData = createSelector(
+  getAllPolicies,
+  getAllPrograms,
+  (policies, programs) => {
+    const data = [];
+
+    if (!policies || !programs) return;
+
+    policies.forEach(p => {
+      const matchingPolicies = programs.filter(
+        program => program.policy === p.policy_id
+      );
+      const matchingGovernments = groupByEntity(matchingPolicies);
+      const totalPolicyCount = Object.keys(matchingGovernments).length;
+      data.push({
+        policy: p.policy_id,
+        policy_name: p.policy_type,
+        policy_desc: p.description,
+        governments: totalPolicyCount,
+      });
+    });
+
+    return data;
+  }
+);
+
+export const getSelectedPolicyData = createSelector(
+  getAllPolicies,
+  getAllPrograms,
+  getSelectedPolicy,
+  (policies, programs, selectedPolicy) => {
+    if (!policies || !programs || !selectedPolicy) return;
+
+    const implementingGovernments = programs.filter(
+      program => program.policy === selectedPolicy
+    );
+
+    const selectedPolicyData = policies.filter(
+      policy => policy.policy_id === selectedPolicy
+    );
```

Comment:
You could use `policies.find` here instead. Then you won't have to do all the `selectedPolicyData[0]` stuff below.